src/desktop/components/follow_button/view.coffee

```diff
@@ -56,7 +56,13 @@ module.exports = class FollowButton extends Backbone.View
       mediator.trigger 'open:auth',
         mode: 'register'
         copy: "Sign up to follow #{@label}"
-        destination: @href || "#{@model.href()}/follow"
+        destination: @href
+        afterSignUpAction: {
```

Comment:
This is the key, we want to store in the cookie the action/id and 'type' (since you can follow different types of things).